builtins/src/main/java/org/jline/builtins/PosixCommands.java

```diff
@@ -1303,6 +1303,9 @@ public static void grep(Context context, String[] argv) throws Exception {
             "  -w --word-regexp         Select only whole words",
             "  -x --line-regexp         Select only whole lines",
             "  -c --count               Only print a count of matching lines per file",
+            "  -z --zero                When used with -c, don't print a count of zero",
+            "  -h --header              Display filename header for each file (defaults to true if multiple files are given)",
```

Comment:
This is usually `-H, --with-filename` and `-h, --no-filename` in `grep`, so I'd rather align with the usual option names.